src/gl/directx11/vcShader.cpp

```diff
@@ -196,14 +196,14 @@ void vcShader_DestroyShader(vcShader **ppShader)
 
 bool vcShader_Bind(vcShader *pShader)
 {
-  if (pShader == nullptr)
-    return false;
-
-  g_pd3dDeviceContext->IASetInputLayout(pShader->pLayout);
+  if (pShader != nullptr)
+  {
+    g_pd3dDeviceContext->IASetInputLayout(pShader->pLayout);
 
-  g_pd3dDeviceContext->VSSetShader(pShader->pVertexShader, NULL, 0);
-  g_pd3dDeviceContext->PSSetShader(pShader->pPixelShader, NULL, 0);
-  g_pd3dDeviceContext->GSSetShader(pShader->pGeometryShader, NULL, 0);
+    g_pd3dDeviceContext->VSSetShader(pShader->pVertexShader, NULL, 0);
+    g_pd3dDeviceContext->PSSetShader(pShader->pPixelShader, NULL, 0);
+    g_pd3dDeviceContext->GSSetShader(pShader->pGeometryShader, NULL, 0);
+  }
 
   return true;
```

Comment:
If this function always returns `true`, why have a return at all?

Reply:
Some graphics APIs can detect an erroneous state (openGL). Having this not 'fail' on `nullptr` is correct for our usage (we use vcShader_Bind(nullptr) for unbinding, however I'm unsure what to do here in d3d yet.)